desktop-src/winmsg/window-features.md

```diff
@@ -162,18 +162,35 @@ The user sets the foreground window by clicking a window, or by using the ALT+TA
 
 An application sets the foreground window by using the [**SetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-setforegroundwindow) function.
 
-The system restricts which processes can set the foreground window. A process can set the foreground window only if one of the following conditions is true:
-
--   The process is the foreground process.
--   The process was started by the foreground process.
--   The process received the last input event.
--   There is no foreground process.
--   The foreground process is being debugged.
--   The foreground is not locked (see [**LockSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-locksetforegroundwindow)).
--   The foreground lock time-out has expired (see **SPI\_GETFOREGROUNDLOCKTIMEOUT** in [**SystemParametersInfo**](/windows/win32/api/winuser/nf-winuser-systemparametersinfoa)).
--   No menus are active.
-
-A process that can set the foreground window can enable another process to set the foreground window by calling the [**AllowSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-allowsetforegroundwindow) function, or by calling the [**BroadcastSystemMessage**](/windows/win32/api/winuser/nf-winuser-broadcastsystemmessage) function with the **BSF\_ALLOWSFW** flag. The foreground process can disable calls to [**SetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-setforegroundwindow) by calling the [**LockSetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-locksetforegroundwindow) function.
+The system restricts which applications can set the foreground window.
+A process can set the foreground window by calling [**SetForegroundWindow**](/windows/win32/api/winuser/nf-winuser-setforegroundwindow) only if:
+
+- At least one of the following conditions is true:
+  - The calling process is the foreground process -
+    the process that created the previous foreground thread
+    and foreground window.
+  - The calling process was started by the foreground process.
+  - There is currently no foreground window, and thus no foreground process.
+  - The calling process received the last input event.
+  - Either the foreground process or the calling process is being debugged.
+- And all of the following conditions are true:
+  - The calling process belongs to a desktop application,
```

Comment:
Please remove this item - adding a functional change like this requires more verification and testing.

Reply:
This is a copy and correction (foreground process --> calling process) of this clause that already exists in the Remarks section for [SetForegroundWindow()](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow) and AllowSetForegroundWindow()'s doc pages:

> The foreground process is not a Modern Application or the Start Screen.

Assuming we'd like to keep this conceptual topic in sync with those API reference pages, there are a few things we can do here:

1. Keep my current addition here.
1. Delete the added clause from here, and from SetForegroundWindow and AllowSetForegroundWindow too. (I can do that as part of my companion PR https://github.com/MicrosoftDocs/sdk-api/pull/394 .)
1. Keep my current addition here, and then change SetForegroundWindow and AllowSetForegroundWindow to remove the "who can set foreground?" remarks entirely and link to the discussion here.
1. Delete the added clause here, and then remove "who can set foreground?" discussion from the API function pages like in the previous option.

What would you suggest? I'm leaning toward the last one.

Reply:
@metathinker I appreciate your points, but I'd feel more comfortable if I could get confirmation from a current MSFT employee. I see that you used to work here - is there you recommend I speak to for confirmation?

Reply:
@hickeys Check the ntuser@microsoft.com alias if it's still around. Otherwise, I forget who exactly owns user32.dll (Alex Snitkovsky?), but you should be able to look them up on the ownership system (https://osgwiki.com/wiki/Ownership ?).